py-polars/polars/series/series.py

```diff
@@ -6821,20 +6821,22 @@ def struct(self) -> StructNameSpace:
         return StructNameSpace(self)
 
 
-def _resolve_datetime_dtype(
-    dtype: PolarsDataType | None, ndtype: np.datetime64
+def _resolve_temporal_dtype(
+    dtype: PolarsDataType | None,
+    ndtype: np.dtypes.TimeDelta64DType | np.dtypes.DateTime64DType,
 ) -> PolarsDataType | None:
-    """Given polars/numpy datetime dtypes, resolve to an explicit unit."""
+    """Given polars/numpy temporal dtypes, resolve to an explicit unit."""
+    PolarsType = Duration if ndtype.type == np.timedelta64 else Datetime
```

Comment:
could the resolution be preserved for durations too?

- np.timedelta64[ms] -> Duration('ms')
- np.timedelta64[us] -> Duration('us')
- (same for 'ns')
?